torch_geometric/data/feature_store.py

```diff
@@ -0,0 +1,354 @@
+    all the parameters necessary to uniquely identify a tensor from the feature
+    store.
+
+    Note that the order of the attributes is important; this is the order in
+    which attributes must be provided for indexing calls. Feature store
+    implementor classes can define a different ordering by overriding
+    TensorAttr.__init__.
+    """
+
+    # The type of the nodes that the tensor corresponds to (may be used for
+    # hetereogeneous graphs)
+    group_name: Optional[str] = _field_status.UNSET
+
+    # The name of the feature tensor (may be used if there are multiple
+    # different feature tensors for the same node index)
+    attr_name: Optional[str] = _field_status.UNSET
+
+    # The node indices the rows of the tensor correspond to
+    index: Optional[IndexType] = _field_status.UNSET
+
+    # Convenience methods #####################################################
+
+    def is_set(self, attr):
+        r"""Whether an attribute is set in :obj:`TensorAttr`."""
+        assert attr in self.__dataclass_fields__
+        return getattr(self, attr) != _field_status.UNSET
+
+    def is_fully_specified(self):
+        r"""Whether the :obj:`TensorAttr` has no unset fields."""
+        return all([
+            getattr(self, field) != _field_status.UNSET
+            for field in self.__dataclass_fields__
+        ])
+
+    def update(self, attr: 'TensorAttr'):
+        r"""Updates an :obj:`TensorAttr` with set attributes from another
+        :obj:`TensorAttr`."""
+        for field in self.__dataclass_fields__:
+            val = getattr(attr, field)
+            if val != _field_status.UNSET:
+                setattr(self, field, val)
+
+
+class AttrView(CastMixin):
+    r"""Defines a view of a :obj:`FeatureStore` that is obtained from a
+    specification of attributes on the feature store. The view stores a
+    reference to the backing feature store as well as a :obj:`TensorAttr`
+    object that represents the view's state.
+
+    Users can create views either using the :obj:`AttrView` constructor,
+    :obj:`FeatureStore.view`, or by incompletely indexing a feature store.
+    """
+    _store: 'FeatureStore'
+    _attr: TensorAttr
+
+    def __init__(self, store: 'FeatureStore', attr: TensorAttr):
+        self._store = store
+        self._attr = attr
+
+    # Python built-ins ########################################################
+
+    def __getattr__(self, key) -> 'AttrView':
+        r"""Sets the attr_name field of the backing :obj:`TensorAttr` object to
+        the attribute. This allows for :obj:`AttrView` to be indexed by
+        different values of attr_name. In particular, for a feature store that
+        has `feat` as an `attr_name`, the following code indexes into `feat`:
+
+        .. code-block:: python
+
+            store[group_name].feat[:]
+
+        """
+        if key in ['_attr', '_store']:
+            return super(AttrView, self).__getattribute__(key)
```

Comment:
Not sure why we need this.
```suggestion
            return getattr(self, key)
```
should work as well.